core/src/main/java/se/tre/freki/search/IdChangeIndexerListener.java

```diff
@@ -49,11 +50,15 @@ public IdChangeIndexerListener(final Store store,
   @AllowConcurrentEvents
   public final void recordLabelCreated(final LabelCreatedEvent event) {
     addCallback(store.getMeta(event.getId(), event.getType()),
-        new FutureCallback<LabelMeta>() {
+        new FutureCallback<Optional<LabelMeta>>() {
           @Override
-          public void onSuccess(final LabelMeta meta) {
-            LOG.info("Indexing {}", meta);
-            addCallback(searchPlugin.indexLabelMeta(meta), pluginError);
+          public void onSuccess(final Optional<LabelMeta> meta) {
+            if (meta.isPresent()) {
+              LOG.info("Indexing {}", meta.get());
+              addCallback(searchPlugin.indexLabelMeta(meta.get()), pluginError);
+            }
+            LOG.error("Unable to fetch LabelMeta object for {}[{}]",
```

Comment:
Can you lower the message level to `warn` or even `info`? Not having a `LabelMeta` here could be normal in certain cases